src/ol/render/canvas/replaygroup.js

```diff
@@ -109,16 +183,19 @@ ol.render.canvas.ReplayGroup.prototype.finish = function() {
  * @template T
  */
 ol.render.canvas.ReplayGroup.prototype.forEachFeatureAtCoordinate = function(
-    coordinate, resolution, rotation, skippedFeaturesHash, callback) {
+    coordinate, resolution, rotation, hitTolerance, skippedFeaturesHash, callback) {
 
-  var transform = ol.transform.compose(this.hitDetectionTransform_,
-      0.5, 0.5,
+  hitTolerance = Math.round(hitTolerance);
+  var contextSize = hitTolerance * 2 + 1;
+  var transform = ol.transform.compose(ol.transform.create(),
+      hitTolerance + 0.5, hitTolerance + 0.5,
       1 / resolution, -1 / resolution,
       -rotation,
       -coordinate[0], -coordinate[1]);
-
   var context = this.hitDetectionContext_;
-  context.clearRect(0, 0, 1, 1);
+  context.canvas.width = contextSize;
+  context.canvas.height = contextSize;
+  context.clearRect(0, 0, contextSize, contextSize);
```

Comment:
For performance reasons, ou should either set  `width` and `height` on the canvas, or call `clearRect()`. Not both. My recommendation for best performance would be to only set `width` and `height` when `contextSize` is different than the current `width` and `height`, and call `clearRect()` otherwise. This is also what we do elsewhere.